Documentation/ApiOverview/UpdateWizards/Creation/Index.rst

```diff
@@ -0,0 +1,74 @@
+.. include:: ../../../Includes.txt
+
+.. _update-wizards-creation-generic:
+
+===============================
+Creating generic update wizards
+===============================
+
+Each update wizard consists of a single PHP file containing a single PHP class. This
+class has to extend :php:`TYPO3\CMS\Install\Updates\AbstractUpdate` and implement the
+abstract methods::
+
+   <?php
+   namespace Vendor\ExtName\Updates;
+
+   use TYPO3\CMS\Install\Updates\AbstractUpdate;
+
+   class ExampleUpdateWizard extends AbstractUpdate
+   {
+       /**
+        * @var string
+        */
+       protected $title = 'Title of this updater';
+
+       /**
+        * Checks whether updates are required.
+        *
+        * @param string &$description The description for the update
+        * @return bool Whether an update is required (TRUE) or not (FALSE)
+        */
+       public function checkForUpdate(&$description)
+       {
+           return true;
+       }
+
+      /**
+       * Performs the accordant updates.
+       *
+       * @param array &$dbQueries Queries done in this update
+       * @param string &$customMessage Custom message
+       * @return bool Whether everything went smoothly or not
+       */
+       public function performUpdate(array &$databaseQueries, &$customMessage)
+       {
+           return true;
+       }
+   }
+
+Property :php:`$title`
+   Can be overwritten to define the title used while rendering the list of available
+   update wizards.
+
+Method :php:`checkForUpdate`
+   Is called to check whether the updater has to run. Therefore a boolean has to be
+   returned. The :php:`$description` provided can be modified as a reference to
+   provide further explanation, in addition to the title.
+
+Method :php:`performUpdate`
+   Is called if the user triggers the wizard. This method should contain, or call,
+   the code that is needed to execute the update. You can use the
+   :php:`$databaseQueries` and :php:`$customMessage` as referenced to provide further
+   information to the user after the update function is completed.
+   :php:`$databaseQueries` has to be an array where each value is a string containing
+   the query. This array should only contain executed queries. :php:`$customMessage`
+   is a single string where you can provide further information to the user.
+
+Marking wizard as done
+======================
+
+As soon as the wizard is fully finished, e.g. it detected that no update is
+necessary, or the update was finished, the wizard should be marked as done.
```

Comment:
"As soon as [...] **_has completely finished_**, e.g. [...] no update is necessary **_anymore_**, or **_that all_** ~the~ updates **_were completed successfully_**, [...]"

(styling only to indicate changes)